.github/workflows/e2e_scripts/cleanup_monitoring.py

```diff
@@ -0,0 +1,121 @@
+import sys
+
+from google.cloud import monitoring_v3
+from google.cloud.monitoring_dashboard import v1
+
+project_id = ''
+project_name = 'projects/'
+
+def getProjectId():
+    """Retrieves the project id from the script arguments.
+    Returns:
+        str -- the project name
+    Exits when project id is not set
+    """
+    try:
+        project_id = sys.argv[1]
+    except:
+        exit('Missing Project ID. Usage: python3 cleanup_monitoring.py $PROJECT_ID')
+
+    return project_id
+
+def cleanupDashboards():
+    """ Deletes all dashboards. """
+    client = v1.DashboardsServiceClient()
+    dashboards = list(client.list_dashboards(project_name))
+    while (len(dashboards) != 0):
+        for dashboard in dashboards:
+            try:
+                client.delete_dashboard(dashboard.name)
+            except:
+                print('Could not delete dashboard: ' + dashboard.name)
+        dashboards = list(client.list_dashboards(project_name))
+
+def cleanupPolicies():
+    """ Delete all alerting policies for both uptime checks and SLOs. """
+    client = monitoring_v3.AlertPolicyServiceClient()
+    policies = list(client.list_alert_policies(project_name))
+    while (len(policies) != 0):
+        for policy in policies:
+            try:
+                client.delete_alert_policy(policy.name)
+            except:
+                print('Could not delete alerting policy: ' + policy.name)
+        policies = list(client.list_alert_policies(project_name))
+
+def cleanupNotificationChannels():
+    """ Deletes all notification channels. """
+    client = monitoring_v3.NotificationChannelServiceClient()
+    channels = list(client.list_notification_channels(project_name))
+    while (len(channels) != 0):
+        for channel in channels:
+            try:
+                client.delete_notification_channel(channel.name)
+            except:
+                print('Could not delete notification channel: ' + channel.name)
+        channels = list(client.list_notification_channels(project_name))
+
+def cleanupServices():
+    """ Deletes only custom services that are identified by a trailing 'srv' in the name. """
+    client = monitoring_v3.ServiceMonitoringServiceClient()
+    # only delete custom services
+    custom_services = [s for s in client.list_services(project_name) if s.name.endswith("srv")]
+    while (len(custom_services) != 0):
+        for service in custom_services:
+            try:
+                client.delete_service(service.name)
+            except:
+                print('Could not delete service: ' + service.name)
+        custom_services = [s for s in client.list_services(project_name) if s.name.endswith("srv")]
+
+def cleanupSlos():
+    """ Deletes every SLO associated with every service. """
+    client = monitoring_v3.ServiceMonitoringServiceClient()
+    slos = [slo for service in client.list_services(project_name) for slo in client.list_service_level_objectives(service.name)]
+    while (len(slos) != 0):
```

Comment:
FYI, a list evaluates to true if it has elements, and false if it's empty. This way is [considered more Pythonic](https://www.techiedelight.com/check-list-empty-python/): `while slos:`.

But it may be slightly less readable to non-Python people. Don't feel like you need to change these

Reply:
you should also avoid using parentheses for single-expression statements in Python

Reply:
I made the change - I don't think a lot of people will be modifying the clean script. Removed extra parentheses :)